### DISCUSSION TITLE
Blocage IP

### DISCUSSION THREAD

**Discussion by toon7272** [bokub/conso-api#46] [Aide]
Bonjour,
Désolé de venir vous solliciter, mais suite à une erreure de codage pour profiter de votre API pour le compteur Linky, j'ai bloqué mon IP en faisant une requête trop importante. Et du coup en dépassant la limite. :(
Pouvez-vous débloquer mon IP svp? ([88.171.50.XXX])
J'ai corrigé mon erreur de script pour que ça ne se reproduise plus. (A trop vouloir automatiser on fait parfois des erreurs.)
Encore désolé pour ça et merci de votre compréhension.
Belle journée
Anthony

**Comment by toon7272** [comment]
Bonsoir,
J'aurai du lire la page html du message d'erreur :)

"What can I do to resolve this?
You can email the site owner to let them know you were blocked. Please include what you were doing when this page came up and the **Cloudflare Ray ID** found at the bottom of this page."

Du coup ci-dessous l'information demandée:

Cloudflare Ray ID: 8715437ccec66fac

Si vous pouvez me débloquer... Merci beaucoup.

Par ailleurs, y'a t'il un autre canal pour discuter? Je viens de faire la création d'un compte entreprise chez Enedis, ayant moi aussi des biens à gérer, et je vais regarder pour utiliser également l'Open-API. J'aurai aimé vous demander quelques conseils avant de commencer à developper mon interface.

Par avance merci pour les deux demandes.
Belle soirée
Anthony

**Reply by toon7272** [reply]
Bonsoir, 
je me permets de revenir pour demander votre clémence. Comme informé précedemment j'ai corrigé mon script pour ne faire que trois demandes par jour

${LINKY} maxpower --format csv --quiet
${LINKY} daily --format csv --quiet
${LINKY} loadcurve --format csv --quiet

Pour l'historique je charge maintenant les exports des fichiers depuis enedis pour les charger en une fois.

Merci de bien vouloir débloquer mon IP.

Belle soirée
Anthony.

**Comment by bokub** [comment]
Salut Anthony

Dis donc c'est une sacré erreur de code que vous avez fait, **2177 requêtes en 12 heures** ! Heureusement que j'insiste sur le fait qu'il faut être raisonnable, sinon ça m'aurait coûté cher en serveurs

En un sens c'est un peu dommage, 50 petites requêtes de plus et vous pouviez dépasser le champion des spammeurs, là vous devrez vous contenter d'une médaille d'argent

**Reply by toon7272** [reply]
Bonsoir,
C'est un fait, dans mon empressement à utiliser l'API je n'avais pas compris les limitations et j'ai donc codé pour récuperer les informations de ma ligne. Ce n'est qu'à posteriori que je me suis rendu compte de mon erreur de codage et que j'aurai du me limiter à juste récuperer les données de la veille.
L'intention n'était pas de spammer mais d'utiliser votre passerelle pour récuperer et analyser mes données.
Par contre normalement depuis deux semaines, vous devriez juste voir 3 requêtes successives quotidienne à 23h02 qui sont bloquées. Je pensai que vous m'excuseriez pour mon erreur et que vous voudriez bien prendre en compte mes excuses.
J'ai supprimé mon job ce jour.
Encore désolé pour le désagrément.

Belle soirée
Anthony

**Reply by bokub** [reply]
Ah je vous ai débloqué cet aprem pourtant

Bonne soirée

**Reply by toon7272** [reply]
Je vous en remercie. Je vais donc me permettre 4 requêtes.
Une manuelle pour confirmer le fonctionnement. Après quoi je rétablirai mon job avec 3 requêtes.
Encore merci pour votre temps à lire mes messages et de m'avoir répondu. Et également de m'avoir débloqué.
Je vous présente encore toutes mes excuses pour la gêne occasionnée.
En vous souhaitant une belle soirée.
Anthony.

**Reply by bokub** [reply]
Merci Anthony, bonne soirée également 🙂

**Comment by toon7272** [comment]
Bonjour, pour information j'ai éffectué le chargement hier des données de la veille à la main. Je viens de remettre en cron le script pour le chargement automatique de la veille. Il tournera ce soir à 23h02. Il y aura 3 requêtes.
Par ailleurs je viens de faire une demande sur le site Enedis pour les données manquantes du 05.04 au 21.04 mais la fonction de téléchargement ne fonctionne plus pour mon espace. J'ai informé Enedis mais en cas de non réponse, ce qui est probable, je me servirai de l'API pour récuperer mes données manquantes. Je m'en tiendrai à 6 requêtes par jour le temps de récuperer les informations manquantes. En espérant que ça ne provoque pas de soucis. Je ne commencerai la récupération que Samedi prochain pour laisser 2 jours pleins à Enedis pour me répondre.
Belle journée
Anthony
